// quisquilia/consts.js
exports.VERSION									= '0.0.7';
exports.APPLICATION_NAME						= 'Zen Bot or Feed';
exports.DEBUG_LEVEL_ERROR						= 0x4;
exports.DEBUG_LEVEL_WARNING						= 0x2;
exports.DEBUG_LEVEL_INFO						= 0x1;

/**
 * if true the service will show more informations in the console
 * @type {Boolean}
 */
exports.DEBUG									= true;
exports.DEBUG_LEVEL								= this.DEBUG_LEVEL_ERROR | this.DEBUG_LEVEL_WARNING | this.DEBUG_LEVEL_INFO;


/**
 * Valid Bot Admins
 * @type {String}
 */
exports.ADMINS									= "an average buu, Estop Revolt, sum27039554@pvp.net, Shêngbô, MrRockitt";

/**
 * [STATUS description]
 * @type {String}
 */
exports.STATUS 									= "<body><profileIcon>28</profileIcon><statusMsg></statusMsg><level>80</level><wins>55555</wins><leaves>11111</leaves><queueType>RANKED_SOLO_5x5</queueType><rankedWins>987653210</rankedWins><rankedLosses>0</rankedLosses><rankedRating>1337</rankedRating><tier>DIAMOND</tier><skinname>Evelynn</skinname><gameStatus>inGame</gameStatus><timeStamp>1317686957096</timeStamp><rankedLeagueName>Jayce's Elite</rankedLeagueName><rankedLeagueTier>DIAMOND</rankedLeagueTier><statusMsg>!help to get to know me</statusMsg></body>";

/**
 * Riot Key
 * @type {String}
 */
exports.RIOT_KEY								= "15347d3b-aa2d-43a8-99c9-12b78a54dbc7";


/**
 * Bot commands
 * @type {String}
 */

//-- admin
exports.COMMAND_JOIN       = "join";
exports.COMMAND_LEAVE      = "part";
exports.COMMAND_CHANNELS   = "channels";

//-- admin
//-- grant/revoke permission to change welcome message
exports.COMMAND_GRANT	  	= "grant";
exports.COMMAND_REVOKE		= "revoke";

//-- normal
exports.COMMAND_HELP       = "help";
exports.COMMAND_ADMIN      = "admin?";
exports.COMMAND_CREATEGAME = "game";
exports.COMMAND_CANCELGAME = "cancel";
exports.COMMAND_LISTGAMES  = "list";
exports.COMMAND_JOINGAME   = "joingame";
exports.COMMAND_SEEN       = "seen";
exports.COMMAND_GLOBAL     = "global";
exports.COMMAND_INV        = "inv";
exports.COMMAND_STATS      = "stats";
exports.COMMAND_INHOUSE    = "inhouse";
exports.COMMAND_INVITEME   = "inviteme";
exports.COMMAND_GREETING   = "greeting";
exports.COMMAND_GREETREMOVE= "greetremove";

//-- easter egg
exports.COMMAND_FORTUNECOOKIE = "fortunecookie";

/**
 *  TABLE Helpers
 */
exports.TABLE_NEWLINE							= String.fromCharCode( 0x2063 ) + "\n";
exports.TABLE_HEADER							= "|---/¯`· .  ¸    .  o °|------------------------------------\n";
exports.TABLE_FOOTER							= "";

/**
 * Help strings for bot commands
 * @type {String}
 */
exports.COMMAND_HELP_ADMIN_DESCRIPTION			= "\"" + this.APPLICATION_NAME + " " + this.VERSION + " |Admin Commands|\n";
exports.COMMAND_HELP_ADMIN_DESCRIPTION			+= this.COMMAND_ADMIN + ", ";
exports.COMMAND_HELP_ADMIN_DESCRIPTION			+= this.COMMAND_JOIN + ", ";
exports.COMMAND_HELP_ADMIN_DESCRIPTION			+= this.COMMAND_CHANNELS + ", ";
exports.COMMAND_HELP_ADMIN_DESCRIPTION			+= this.COMMAND_LEAVE;

exports.COMMAND_HELP_DESCRIPTION				= this.TABLE_NEWLINE;
exports.COMMAND_HELP_DESCRIPTION				+= this.TABLE_HEADER + "\n";
exports.COMMAND_HELP_DESCRIPTION				+= "" + this.APPLICATION_NAME + ". " + this.VERSION + "\n\n";
exports.COMMAND_HELP_DESCRIPTION				+= "" + this.APPLICATION_NAME + " manages your channel games, by keeping";
exports.COMMAND_HELP_DESCRIPTION				+= " track of every active game created and their open";
exports.COMMAND_HELP_DESCRIPTION				+= " spots. Any user can create a game just by writing";
exports.COMMAND_HELP_DESCRIPTION				+= " at least 2 of the following info: spots, type and";
exports.COMMAND_HELP_DESCRIPTION				+= " map. For example the phrase 'im creating a blind";
exports.COMMAND_HELP_DESCRIPTION				+= " game with +4' will register a new game. For more";
exports.COMMAND_HELP_DESCRIPTION				+= " info visit the url: http://www.reddit.com/r/zenlol/\n\n";
exports.COMMAND_HELP_DESCRIPTION				+= "Commands: \n";
exports.COMMAND_HELP_DESCRIPTION				+= "" + "!"+this.COMMAND_HELP + ", " + "!"+this.COMMAND_LISTGAMES + ", " + "!"+this.COMMAND_SEEN + ", " + "!"+this.COMMAND_CANCELGAME + ", "  + "!"+this.COMMAND_GLOBAL + ", "  + "!"+this.COMMAND_INV + ", " + "!"+this.COMMAND_STATS + ", " + "!" + this.COMMAND_GREETING + ", " + "!" + this.COMMAND_GREETREMOVE + "\n\n";
exports.COMMAND_HELP_DESCRIPTION				+= "'!help command' will give information about the command";
exports.COMMAND_HELP_DESCRIPTION				+= this.TABLE_FOOTER;


exports.COMMAND_JOIN_DESCRIPTION       = "!join <channel name> - The bot will join the desired channel.";
exports.COMMAND_LEAVE_DESCRIPTION      = "!part <channel name> - The bot will part the desired channel. If no channel is given the bot will depart the current channel.";
exports.COMMAND_ADMIN_DESCRIPTION      = "";
exports.COMMAND_CREATEGAME_DESCRIPTION = "";
exports.COMMAND_LISTGAMES_DESCRIPTION  = "!list - Current available games";
exports.COMMAND_JOINGAME_DESCRIPTION   = "";
exports.COMMAND_SEEN_DESCRIPTION       = "!seen <player> - last time player was on the channel";
exports.COMMAND_CANCELGAME_DESCRIPTION = "!cancel - cancel your current game, if it exists.";
exports.COMMAND_GLOBAL_DESCRIPTION     = "!global - when ON you can see all the games registered, when OFF you can see only the games registered in the channel you asked.";
exports.COMMAND_INV_DESCRIPTION        = "!inv #game_number - the bot will tell the creator of the game you requested to invite you";
exports.COMMAND_STATS_DESCRIPTION      = "!stats - the bot will show general stats";
exports.COMMAND_GREETING_DESCRIPTION	= "!greeting <text> - set the greeting you want the bot to say when you join (Just typing !greeting will tell you the current greeting you have set.)";
exports.COMMAND_GREETREMOVE_DESCRIPTION = "!greetremove - Zen Bot drinks until he forgets all about you and he won't greet you the next time you join.";


exports.ADVERTISEMENT                  = this.APPLICATION_NAME + " " + this.VERSION + " Gaming Bot - type !help for info.";


exports.QUOTES =
[
"As balance dictates.",
"You wanna play, too? It'll be fun!",
"ok.",
"Let's do this.",
"Smart choice!",
"Lady Luck is smilin",
"Don't you find me beautiful?",
"How about a drink?",
"I knew you would do that.",
"I'm up to snuff, I gots me an ace machine!",
"On my wings.",
"This'll be a slaughter!",
"Mundo!",
"The night is my veil.",
"Time for a true display of skill.",
"We'll bring them pain!",
"Your bidding, master?",
"A guardian is always prepared.",
"To the Fields of Justice!",
"More than just precious stones, I bring you an ancient power.",
"If you're buying, I'm in!",
"You'd wish the world you know to end! Yeeeesssss..",
"Indeed, a wise choice!",
"My blade is at your service.",
"The tempest is at your command.",
"How about a magic trick?",
"Into the fray.",
"Violence solves everything.",
"The eyes never lie.",
"Time to feast!",
"The Black Rose shall bloom once more.",
"Do you feel a chill?",
"Tactical decision, summoner!",
"Rock Solid!",
"Oblivion awaits.",
"My blade is yours.",
"Nothing can hold me back!",
"Fortune doesn't favour fools.",
"I shall bring great suffering.",
"The cycle of life and death continues, we will live, they will die.",
"They will fear the wild.",
"Leave nothing behind.",
"They are priviledged to die at my feet.",
"Prepare to be boarded!",
"Valoran will know harmony.",
"Let's go, let's GO!",
"I thought you'd never pick me...",
"A demonstration of superior judgement.",
"I do it.",
"To battle!",
"Only you can hear me summoner, what masterpiece shall we play today?",
"Let me guide you.",
"Fired up and ready to serve.",
"The early bird GUTS the worm.",
"Captain Teemo on duty!",
"I wanna shoot something!",
"Time to troll!",
"I'm on the case.",
"By my will, this shall be finished!",
"An enlightened decision.",
"The balance of power must be preserved.",
"I do your bidding, for now.",
"Embrace the darkness!",
"Here we go!",
"As I live, all will die!",
"Oh, what do you want?",
"Our rage is beyond your control.",
"Eternal life, endless torture.",
"The rivers will run red.",
"To the arena!",
"Let's make this fun!",
"Know that if the tables were turned, I would show you no mercy!",
"Your will, my hands."
];

exports.DISCQUOTES =
[
"If complete and utter chaos was lightning, then he'd be the sort to stand on a hilltop in a thunderstorm wearing wet copper armour and shouting 'All gods are bastards'.",
"What he didn't like about heroes was that they were usually suicidally gloomy when sober and homicidally insane when drunk.",
"The Watch were always careful not to intervene too soon in any brawl where the odds were not heavily stacked in their favour. The job carried a pension, and attracted a cautious, thoughtful kind of man.",
"'It could be worse,' he said by way of farewell. 'It could be me.'",
"That's what's so stupid about the whole magic thing, you know. You spend twenty years learning the spell that makes nude virgins appear in your bedroom, and then you're so poisoned by quicksilver fumes and half-blind from reading old grimoires that you can't remember what happens next.",
"At the back of his mind a bad feeling began to grow. He thought about how it might be to be, say, a fox confronted with an angry sheep. A sheep, moreover, that could afford to employ wolves.",
"Picturesque meant - he decided after careful observation of the scenery that inspired Twoflower to use the word - that the landscape was horribly precipitous. Quaint, when used to describe the occasional village through which they passed, meant fever-ridden and tumbledown. Twoflower was a tourist, the first ever seen on the Discworld. Tourist, Rincewind had decided, meant 'idiot'.",
"It was all very well going on about pure logic and how the universe was ruled by logic and the harmony of numbers, but the plain fact of the matter was that the Disc was manifestly traversing space on the back of a giant turtle and the gods had a habit of going round to atheists' houses and smashing their windows.",
"'You know, I never imagined there were he-dryads. Not even in an oak tree.'",
"One of the giants grinned at him.",
"Druellae snorted. 'Stupid! Where do you think acorns come from?'",
"What heroes like best is themselves.",
"We've strayed into a zone with a high magical index...",
"'We've strayed into a zone with a high magical index,' he said. 'Don't ask me how. Once upon a time a really powerful magic field must have been generated here, and we're feeling the after-effects.'",
"'Precisely,' said a passing bush.",
"The only reason for walking into the jaws of Death is so's you can steal his gold teeth.",
"'It is forbidden to fight on the Killing Ground,' he said, and paused while he considered the sense of this. 'You know what I mean, anyway...'",
"Ripples of paradox spread out across the sea of causality.",
"He wondered what kind of life it would be, having to keep swimming all the time to stay exactly in the same place. Pretty similar to his own, he decided.",
"I've seen excitement, and I've seen boredom. And boredom was best.",
"Some pirates achieved immortality by great deeds of cruelty or derring-do. Some achieved immortality by amassing great wealth. But the captain had long ago decided that he would, on the whole, prefer to achieve immortality by not dying.",
"It was octarine, the colour of magic. It was alive and glowing and vibrant and it was the undisputed pigment of the imagination, because wherever it appeared it was a sign that mere matter was a servant of the powers of the magical mind. It was enchantment itself.",
"But Rincewind always thought it looked a sort of greenish-purple.",
"I've seen excitement, and I've seen boredom. And boredom was best.",
"'What's this wine — crushed octopus eyeballs?'",
"'Sea grape,' said the old man.",
"'Great,' said Rincewind, and swallowed a glassful. 'Not bad. A bit salty, maybe.'",
"'Sea grape is a kind of small jellyfish,' explained the stranger. '[...] Why has your friend gone that strange colour?'",
"'Culture shock, I imagine,' said Twoflower.",
"'We don't have gods where I come from,' said Twoflower.",
"'You do, you know,' said the Lady. 'Everyone has gods. You just don't think they're gods.'",
"'[...] on the disc, the Gods are not so much worshipped as blamed.'",
"Don’t ask me how I knew—I suppose it was because it was just about the worst possible thing that was likely to happen.",
"A Thaum is the basic unit of magical strength. It has been universally established as the amount of magic needed to create one small white pigeon or three normal-sized billiard balls.",
"Pratchett here draws upon the Greek thauma or 'marvel' in creating the name of this basic unit.",
"The sun rose slowly, as if it wasn't sure it was worth all the effort.",
"The pen is mightier than the sword ... if the sword is very short, and the pen is very sharp.",
"The disc, being flat, has no real horizon. Any adventurous sailor who got funny ideas from staring at eggs and oranges for too long and set out for the antipodes soon learned that the reason why distant ships sometimes looked as though they were disappearing over the edge of the world was that they were disappearing over the edge of the world.",
"Of course I'm sane, when trees start talking to me, I don't talk back.",
"Weems might have had a room-temperature IQ, but he knew idiocy when he saw it.",
"Early to rise, early to bed, makes a man healthy, wealthy and dead.",
"'Dead?' said Rincewind, In the debating chamber of his mind a dozen emotions got to their feet and started shouting. Relief was in full spate when Shock cut in on a point of order and then Bewilderment, Terror and Loss started a fight which was ended only when Shame slunk in from next door to see what all the row was about.",
"Darkness isn't the opposite of light, it is simply its absence.",
"Ankh-Morpork!",
"Pearl of cities!",
"This is not a completely accurate description, of course — it was not round and shiny — but even its worst enemies would agree that if you had to liken Ankh-Morpork to anything, then it might as well be a piece of rubbish covered with the diseased secretions of a dying mollusc.",
"The death of the warrior or the old man or the little child, this I understand, and I take away the pain and end the suffering. I do not understand this death-of-the-mind.",
"Here is the blackness of space, the myriad stars gleaming like diamond dust or, as some people would say, like great balls of exploding hydrogen a very long way off. But then, some people would say anything.",
"- 'If you're going to suggest I try dropping twenty feet down a pitch dark tower in the hope of hitting a couple of greasy little steps which might not even still be there, you can forget it,' said Rincewind sharply.",
"- 'There is an alternative, then.'",
"- 'Out with it, man.'",
"- 'You could drop five hundred feet down a pitch black tower and hit stones which certainly are there,' said Twoflower.",
"Radiating from the book was the light that lies on the far side of darkness, the light fantastic.",
"It was a rather disappointing purple colour.",
"This is a story about magic and where it goes and perhaps more importantly where it comes from and why, although it doesn't pretend to answer all or any of these questions.",
"It may, however, help to explain why Gandalf never got married and why Merlin was a man. Because this is also a story about sex, although probably not in the athletic, tumbling, count-the-legs-and-divide-by-two sense unless the characters get totally beyond the author's control. They might.",
"A world like that, which exists only because the gods enjoy a joke, must be a place where magic can survive. And sex too, of course.",
"It wasn't a large village, and wouldn't have shown up on a map of the mountains. It barely showed up on a map of the village.",
"Time passed, which, basically, is its job.",
"'They're both magic. If you can't learn to ride an elephant, you can at least learn to ride a horse.'",
"'What's an elephant?'",
"'A kind of badger,' said Granny. She hadn't maintained forest-credibility for forty years by ever admitting ignorance.",
"'Do you think I used magic?'",
"Esk looked down at the queen bee. She looked up at the witch.",
"'No,' she said, 'I think you just know a lot about bees.'",
"Granny grinned.",
"'Exactly correct. That’s one form of magic, of course.'",
"'What, just knowing things?'",
"'Knowing things that other people don’t know,' said Granny.",
"'Gods are all right,' said Granny, as they ate their lunch and looked at the view. 'You don’t bother gods, and gods don’t come bothering you.'",
"I'm surely going to regret this, she told herself, displaying considerable foresight.",
"There would be a price. And Granny knew enough about wizardry to be certain that it would be a high one. But if you were worried about the price, then why were you in the shop?",
"They say a little knowledge is a dangerous thing, but it is not one half so bad as a lot of ignorance.",
"'You mean it's my destiny?' she said at last.",
"Granny shrugged. 'Something like that. Probably. Who knows?'",
"Although his body had been around quite a lot, his mind had never gone further than the inside of his own head.",
"He had the kind of real deep tan that rich people spend ages trying to achieve with expensive holidays and bits of tinfoil, when really all you need to do to obtain one is work your arse off in the open air every day.",
"She liked it because it offered privacy, always appreciated by, as she put it, 'my more discerning clients who prefer to make their very special purchases in an atmosphere of calm where discretion is ever the watchword.'",
"A hint was to Esk what a mosquito bite was to the average rhino because she was already learning that if you ignore the rules people will, half the time, quietly rewrite them so that they don't apply to you.",
"It is well known that a vital ingredient of success is not knowing that what you're attempting can't be done. A person ignorant of the possibility of failure can be a halfbrick in the path of the bicycle of history.",
"One reason for the bustle was that over large parts of the continent other people preferred to make money without working at all, and since the Disc had yet to develop a music recording industry they were forced to fall back on older, more traditional forms of banditry.",
"Wizards parted with money slightly less readily than tigers parted with their teeth.",
"It had been a very long night, and the morning didn’t seem to be an improvement.",
"The Shades, in brief, were an abode of discredited gods and unlicensed thieves, ladies of the night and peddlers in exotic goods, alchemists of the mind and strolling mummers; in short, all the grease on civilization’s axle.",
"Granny had counted the temples with a thoughtful look in her eyes; gods were always demanding that their followers acted other than according to their true natures, and the human fallout this caused made plenty of work for witches.",
"Granny had nothing against fortune-telling provided it was done badly by people with no talent for it.",
"'Can't you read, Esk?'",
"The astonishment in his voice stung her.",
"'I expect so,' she said defiantly. 'I've never tried.'",
"'Hmm. Granpone the White. He's going to be Granpone the Grey if he doesn't take better care of his laundry. Aye tell you, girl, a white magician is just a black magician with a good housekeeper.'",
"'Million-to-one chances,' she said, 'crop up nine times out of ten.'",
"'Who says that?' said Cutangle.",
"'Generally people who are wrong.' said Granny. 'I makes a note in my Almanack, see. I checks. Most things most people believe are wrong.'",
"She told me that if magic gives people what they want, then not using magic can give them what they need."
];

exports.FORTUNECOOKIE = [
"Being deeply loved by someone gives you strength, while loving someone deeply gives you courage.",
"A journey of a thousand miles begins with a single step.",
"If you do not change direction, you may end up where you are heading.",
"To the mind that is still, the whole universe surrenders.",
"I have just three things to teach: simplicity, patience, compassion. These three are your greatest treasures.",
"Silence is a source of great strength.",
"Knowing others is wisdom, knowing yourself is Enlightenment.",
"Music in the soul can be heard by the universe.",
"A good traveler has no fixed plans, and is not intent on arriving.",
"Life and death are one thread, the same line viewed from different sides.",
"Mastering others is strength. Mastering yourself is true power.",
"Love is of all passions the strongest, for it attacks simultaneously the head, the heart and the senses.",
"Nothing is softer or more flexible than water, yet nothing can resist it.",
"When I let go of what I am, I become what I might be.",
"Anticipate the difficult by managing the easy.",
"The key to growth is the introduction of higher dimensions of consciousness into our awareness.",
"Be content with what you have; rejoice in the way things are. When you realize there is nothing lacking, the whole world belongs to you.",
"The snow goose need not bathe to make itself white. Neither need you do anything but be yourself.",
"He who knows, does not speak. He who speaks, does not know.",
"Man's enemies are not demons, but human beings like himself.",
"He who does not trust enough, Will not be trusted.",
"Be the chief but never the lord.",
"The wise man does not lay up his own treasures. The more he gives to others, the more he has for his own.",
"Nature does not hurry, yet everything is accomplished.",
"Respond intelligently even to unintelligent treatment.",
"Great acts are made up of small deeds.",
"If you realize that all things change, there is nothing you will try to hold on to. If you are not afraid of dying, there is nothing you cannot achieve.",
"A leader leads by example, not by force.",
"Let your plans be dark and impenetrable as night, and when you move, fall like a thunderbolt.",
"Victorious warriors win first and then go to war, while defeated warriors go to war first and then seek to win.",
"The greatest victory is that which requires no battle.",
"In the midst of chaos, there is also opportunity.",
"One may know how to conquer without being able to do it.",
"The expert in battle moves the enemy, and is not moved by him.",
"Attack is the secret of defense; defense is the planning of an attack.",
"One day you will find the right words, and they will be simple.",
"Live, travel, adventure, bless, and don't be sorry.",
"The only truth is music.",
"My fault, my failure, is not in the passions I have, but in my lack of control of them.",
"Great things are not accomplished by those who yield to trends and fads and popular opinion.",
"I have nothing to offer anybody except my own confusion.",
"One man practicing kindness in the wilderness is worth all the temples this world pulls.",
"Rewards and punishments are the lowest form of education.",
"A path is made by walking on it.",
"The wise man knows that it is better to sit on the banks of a remote mountain stream than to be emperor of the whole world.",
"We are born from a quiet sleep, and we die to a calm awakening.",
"After silence, that which comes nearest to expressing the inexpressible is music.",
"Maybe this world is another planet’s hell.",
"If one's different, one's bound to be lonely.",
"An intellectual is a person who has discovered something more interesting than sex.",
"There are things known and there are things unknown, and in between are the doors of perception.",
"That men do not learn very much from the lessons of history is the most important of all the lessons that history has to teach.",
"Experience is not what happens to a man; it is what a man does with what happens to him.",
"The secret of genius is to carry the spirit of the child into old age, which means never losing your enthusiasm.",
"Consistency is contrary to nature, contrary to life. The only completely consistent people are the dead.",
"Happiness is not achieved by the conscious pursuit of happiness; it is generally the by-product of other activities.",
"Never let your sense of morals prevent you from doing what is right.",
"Self-education is, I firmly believe, the only kind of education there is.",
"Your assumptions are your windows on the world. Scrub them off every once in a while, or the light won't come in.",
"In a time of deceit telling the truth is a revolutionary act.",
"He who controls the past controls the future. He who controls the present controls the past.",
"War is Peace. Freedom is Slavery. Ignorance is Strength.",
"All animals are equal, but some animals are more equal than others.",
"Man suffers only because he takes seriously what the gods made for fun",
"This is the real secret of life -- to be completely engaged with what you are doing in the here and now. And instead of calling it work, realize it is play.",
"You are an aperture through which the universe is looking at and exploring itself.",
"It would be possible to describe everything scientifically, but it would make no sense; it would be without meaning, as if you described a Beethoven symphony as a variation of wave pressure."
];
